Assets/MRTK/Core/EventDatum/Input/InputEventData.cs

```diff
@@ -32,7 +37,39 @@ public void Initialize(IMixedRealityInputSource inputSource, Handedness handedne
         {
             BaseInitialize(inputSource, inputAction);
             Handedness = handedness;
+
+            skipElements.Clear();
+
+            Status = LifeStatus.None;
+            Propagation = EventPropagation.BubblesUp | EventPropagation.TricklesDown;
+            Phase = PropagationPhase.None;
         }
+
+        /// <inheritdoc />
+        public void PreventDefault()
+        {
+            Status |= LifeStatus.DefaultPrevented;
+        }
+
+        /// <inheritdoc />
+        public void StopPropagation()
+        {
+            Status |= LifeStatus.PropagationStopped;
+        }
+
+        /// <inheritdoc />
+        public void StopPropagationImmediately()
+        {
+            Status |= LifeStatus.ImmediatePropagationStopped;
```

Comment:
i would rename the status flag to PropagationStoppedImmediately else it suggests we have a immediate propagation and a default one instead of a stop and immediate stop

Reply:
Nice one. Agree.